zerver/decorator.py

```diff
@@ -114,6 +114,11 @@ def wrapper(request, user_profile, *args, **kwargs):
         # type: (HttpRequest, UserProfile, *Any, **Any) -> HttpResponse
         if not user_profile.is_realm_admin:
             raise JsonableError(_("Must be a realm administrator"))
+
+        if not check_subdomain(get_subdomain(request), user_profile.realm.subdomain):
+            logging.warning("User %s attempted to access wrong subdomain %s" % (
+                user_profile.email, get_subdomain(request)))
+            raise JsonableError(_("Account is not associated with this subdomain"))
```

Comment:
Can you explain why this is required?  In order to work, `@require_realm_admin` needs to be used with another authenticate decorator, right?